test/Interop/Cxx/stdlib/use-std-map.swift

```diff
@@ -88,6 +88,30 @@ StdMapTestSuite.test("UnorderedMap.subscript") {
   expectNil(m[-1])
 }
 
+StdMapTestSuite.test("Map.filter") {
+  var m = initMap()
+
+  expectNotNil(m[1])
+
+  m = m.filter { k, v in k != 1 }
+
+  expectNil(m[1])
+  expectEqual(m[2], 2)
+  expectEqual(m[3], 3)
```

Comment:
Could you please also add a test to ensure that calling `.filter` on an empty dictionary produces another empty dictionary?

Reply:
Added tests for `map` and `unordered_map`.
If you have any comments on the empty dictionary test approach, please let me know.